### PR TITLE
Add BufferScope for handling buffer rentals for interop

### PR DESCRIPTION
**Pull request by JeremyKuhne**
`BufferScope` is a ref struct that handles rental and return from the `ArrayPool` as well as:

- Initial stack allocated buffers
- Reacquisition of buffers (with content copying if necessary)
- Span semantics (converters, range operators, etc.)

It significantly simplifies creating temporary buffers, particularly in cases where you need to make multiple calls (for the buffer not being large enough).

Also add `[SkipLocalsInit]` in a few places (so the stack allocated buffer doesn't get zeroed out).

Did a global removal of unused usings (most of the file changes are just this).

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8999)

### CODE REVIEW

File: src/System.Windows.Forms.Primitives/tests/UnitTests/System/BufferScopeTests.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        Assert.Equal("Z", buffer[..1].ToString());
+    }
+
+    [Fact]
+    public void Construct_GrowAndCopy()
+    {
+        using BufferScope<char> buffer = new(32);
+        Assert.True(buffer.Length >= 32);
+        buffer[0] = 'Y';
+        buffer.EnsureCapacity(64, copy: true);
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Primitives/tests/UnitTests/System/BufferScopeTests.cs:50]: The other test is copying already, perhaps this one can e `copy: false`

File: src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2PropertyDescriptor.cs [hunk 2]
```diff
@@ -984,15 +983,13 @@ public override unsafe void SetValue(object? component, object? value)
                     (uint)hr,
                     PInvoke.GetThreadLocale(),
                     b,
-                    255,
+                    (uint)buffer.Length - 2,
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2PropertyDescriptor.cs:986]: Could you please clarify the `-2`, I see that it was used before, but don't know why. This size should be in TCHARs.

**Review by Tanya-Solyanik — DISMISSED** [review]
Looks good except for the tests. Also I only skimmed across the "remove unused usings" part of the change assuming it was done by the analyzer.

File: src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2PropertyDescriptor.cs (hunk 2 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2PropertyDescriptor.cs:986]: I can't clarify as I think the code is actually wrong. I think it is the common "I can't figure out the buffer stuff with StringBuilder so I'll pad all over the place". Not super important that we're wasting a few chars here so I'm not planning to spend time validating.

TCHAR is UTF-16 (`char`). It is only 8bit on Win9x.

File: src/System.Windows.Forms.Primitives/tests/UnitTests/System/BufferScopeTests.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms.Primitives/tests/UnitTests/System/BufferScopeTests.cs:50]: `copy: false` is the default and I want to test in both methods as one has a backing `stackalloc` and the other doesn't (and impacts the copy as a result).

**Review by Tanya-Solyanik — APPROVED** [review]

**Pull request merged**